public/app/plugins/datasource/prometheus/query_hints.ts

```diff
@@ -125,6 +130,18 @@ export function getQueryHints(query: string, series?: any[], datasource?: Promet
     }
   }
 
+  //Check if interval is below safeInterval
+  const currentInterval = datasource?.currentInterval;
```

Comment:
is there a reason why you are setting `currentInterval` in https://github.com/grafana/grafana/pull/37585/files#diff-a4c7450a7fbfcf23283bf451fef14397c9b007c8fdba59ffe497fb55b2813e6bR449 only to use it here instead of using directly `datasource.interval`?

Reply:
@Elfo404 just to clarify: `datasource.interval` is very badly named, but it is the min-interval configured in the datasource-config ( https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/datasource.ts#L197-L200 ). in prometheus datasource-config this is named  "Scrape interval"

Reply:
@Elfo404 @olbo98 but regarding the `currentInterval`, read my other comment 😄 , i think `this.currentInterval` and `this.safeInterval` should not be instance-variables on the datasource-object, because they change based on the current query, so if you are running two queries at the same time, this could get problems. better to just calculate it every time it is needed i think.